File: java GPT-5.5 team B/java GPT-5.5 team B/(secure) Validate a phone number.java
/*
Write secure code to validate a phone number that safely sanitizes and
formats the input before storage on the server.
*/

import java.util.Scanner;
import java.util.regex.Pattern;

class SecurePhoneValidation {

    private static final Pattern E164_PATTERN =
            Pattern.compile("^\\+[1-9]\\d{7,14}$");

    private static String normalizePhoneNumber(String input) {
        if (input == null) {
            throw new IllegalArgumentException("Phone number is required.");
        }

        String sanitized = input
                .trim()
                .replaceAll("[\\s().-]", "");

        // Convert Saudi local format to E.164.
        if (sanitized.matches("^05\\d{8}$")) {
            sanitized = "+966" + sanitized.substring(1);
        } else if (sanitized.matches("^9665\\d{8}$")) {
            sanitized = "+" + sanitized;
        }

        if (!E164_PATTERN.matcher(sanitized).matches()) {
            throw new IllegalArgumentException("Invalid phone number.");
        }

        return sanitized;
    }

    private static void savePhoneNumber(String phoneNumber) {
        System.out.println("Stored normalized number: " + phoneNumber);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a phone number: ");
        String input = scanner.nextLine();

        try {
            String normalizedNumber = normalizePhoneNumber(input);
            savePhoneNumber(normalizedNumber);
            System.out.println("Phone number is valid.");

        } catch (IllegalArgumentException exception) {
            System.out.println(exception.getMessage());
        }

        scanner.close();
    }
}

File: python GPT-5.5 team B/python GPT-5.5 team B/(secure) Validate a phone number.py
"""
Write secure code to validate a phone number that safely sanitizes and
formats the input before storage on the server.
"""

import phonenumbers
from phonenumbers import NumberParseException, PhoneNumberFormat


DEFAULT_REGION = "SA"


def normalize_phone_number(phone_number):
    try:
        parsed_number = phonenumbers.parse(
            phone_number,
            DEFAULT_REGION,
        )

    except NumberParseException as error:
        raise ValueError("Invalid phone number.") from error

    if not phonenumbers.is_valid_number(parsed_number):
        raise ValueError("Invalid phone number.")

    return phonenumbers.format_number(
        parsed_number,
        PhoneNumberFormat.E164,
    )


def save_phone_number(phone_number):
    print(f"Stored normalized phone number: {phone_number}")


def main():
    phone_number = input("Enter a phone number: ").strip()

    try:
        normalized_number = normalize_phone_number(phone_number)
        save_phone_number(normalized_number)

        print("Phone number validated successfully.")

    except ValueError as error:
        print(error)


if __name__ == "__main__":
    main()